client/trino-client/src/main/java/io/trino/client/ClientSession.java

```diff
@@ -34,7 +34,7 @@
 public class ClientSession
 {
     private final URI server;
-    private final String principal;
+    private final Optional<String> principal;
```

Comment:
Why this got optional too?

Reply:
The JDBC driver uses trino client so in order to make it optional in JDBC I had to make it optional in the client first.